solidstart-tanstackquery-tailwind-modules/src/services/get-gists.tsx

```diff
@@ -22,7 +26,31 @@ const getGists = async () => {
   };
 
   const resp = (await FetchApi(data)) as Response;
-  return resp?.data.gists || null;
+
+  const gists = resp.data?.viewer.gists.nodes?.reduce((acc: any, gist) => {
+    if (!gist) {
+      return acc;
+    }
+    const files = gist.files ?? [];
+    const gists = files.reduce(
+      (_acc, file) =>
+        file
+          ? [
+              ..._acc,
+              {
+                id: gist.id,
+                description: gist.description,
+                name: file.name || gist.name,
+                url: gist.url,
+              },
+            ]
+          : acc,
+      []
+    );
+    return [...acc, ...gists];
+  }, []);
```

Comment:
We can type this function as:

```suggestion
  const gists = resp.data?.viewer.gists.nodes?.reduce((acc: Gists[] | { name: string }[], gist) => {
    if (!gist) {
      return acc;
    }
    const files = gist.files ?? [];
    const gists = files.reduce(
      (_acc: { name: string }[], file) =>
        file
          ? [
            ..._acc,
            {
              id: gist.id,
              description: gist.description,
              name: file.name || gist.name,
              url: gist.url,
            },
          ]
          : acc,
      []
    );
    return [...acc, ...gists];
  }, []);
```